src/daft-functions/src/list/unique_count.rs

```diff
@@ -0,0 +1,66 @@
+use std::any::Any;
+
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    prelude::{DataType, Field, Schema},
+    series::Series,
+};
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct ListUniqueCount;
+
+#[typetag::serde]
+impl ScalarUDF for ListUniqueCount {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn name(&self) -> &'static str {
+        "unique_count"
```

Comment:
IIUC this expression is a scalar expression that should only work on lists.

1. Should this be `list_unique_count` instead? 
2. Should we not be performing type validation in`to_field`?

Reply:
Yes, you are right on both! Will update in the coming commit.

Reply:
Updated.